Include/Rover/TrialIterator.hpp

```diff
@@ -0,0 +1,520 @@
+      T>>>::operator -=(std::ptrdiff_t offset) {
+    m_offset -= offset;
+    return *this;
+  }
+
+  template<typename T>
+  TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<T>>>
+      TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator +(std::ptrdiff_t offset) const {
+    auto result = *this;
+    result += offset;
+    return result;
+  }
+
+  template<typename T>
+  TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<T>>>
+      TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator -(std::ptrdiff_t offset) const {
+    auto result = *this;
+    result -= offset;
+    return result;
+  }
+
+  template<typename T>
+  std::ptrdiff_t TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator -(const TrialIterator& other) const {
+    return m_offset - other.m_offset;
+  }
+
+  template<typename T>
+  bool TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator ==(const TrialIterator& other) const {
+    return m_offset == other.m_offset;
+  }
+
+  template<typename T>
+  bool TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator !=(const TrialIterator& other) const {
+    return m_offset != other.m_offset;
+  }
+
+  template<typename T>
+  bool TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator <(const TrialIterator& other) const {
+    return m_offset < other.m_offset;
+  }
+
+  template<typename T>
+  bool TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator <=(const TrialIterator& other) const {
+    return m_offset <= other.m_offset;
+  }
+
+  template<typename T>
+  bool TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator >(const TrialIterator& other) const {
+    return m_offset > other.m_offset;
+  }
+
+  template<typename T>
+  bool TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator >=(const TrialIterator& other) const {
+    return m_offset >= other.m_offset;
+  }
+
+  template<typename T>
+  typename const TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::Sample& TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator *() {
+    m_sample = (*m_trial)[m_offset];
+    return m_sample;
+  }
+
+  template<typename T>
+  typename const TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::Sample* TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator ->() {
+    m_sample = (*m_trial)[m_offset];
+    return std::addressof(m_sample);
+  }
+
+  template<typename T>
+  typename const TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::Sample& TrialIterator<T, std::enable_if_t<returns_sample_by_copy_v<
+      T>>>::operator [](std::ptrdiff_t offset) {
+    m_sample = (*m_trial)[m_offset + offset];
```

Comment:
Use ```emplace``` instead of assignment. Each ```Sample``` should be considered an entirely distinct object rather than the mutation of an existing object and furthermore there is no guarantee that ```Sample``` has an assignment operator.